### DISCUSSION TITLE
Avoid @markdoc/next.js node's transform to run on client

### DISCUSSION THREAD

**Discussion by asaf** [markdoc/markdoc#336] [Q&A] [answered]
Hey,

I'm using the `@markdoc/next.js` and configured a custom `transform` method on the Fence node located in `markdoc/nodes.ts` file (I have some Prism code that doesn't work well on client side and can't use react-prism atm)

Is there a way to ensure `transform` runs only on server side?

Best,
Asaf.

**Comment by mfix-stripe** [comment, marked as answer]
Hey @asaf, `transform` should always only run server-side—it is called within `getStaticProps`.

**Reply by asaf** [reply]
Hey @mfix-stripe

I'm talking about the transform function of nodes, not the markdoc's main transform function.

Thanks

**Reply by mfix-stripe** [reply]
Those also won't get called anywhere except on the server, since the node transform functions get called downstream of Markdoc's main `transform` function: https://github.com/markdoc/markdoc/blob/main/src/transformer.ts#L66

**Comment by asaf** [comment]
I got confused because the error I get from `prismjs` is exactly what I get when I try to import it on the client side,

Is `@markdoc/next.js` package has any kind import limitations when using the `markdoc` folder?

This is the error I get from Prism's `loadLanguages()` func: 
```bash
Error: Cannot find module './prism-markup'
```

I have exactly the same code running within `getStaticProps` or within `next.config.js` and it works just fine.

Here's an example:
https://codesandbox.io/s/frosty-mountain-9l7w8d?file=/markdoc/nodes/fence.markdoc.ts


Best,
Asaf.

**Reply by mfix-stripe** [reply]
@asaf that seems entirely separate from the Markdoc config. I would try and call `loadLanguages` on the frontend, either in your `CodeBlock` file or on mount within `CodeBlock`

**Comment by asaf** [comment]
I know it's behind the scope of Markdoc functionality but import behaves differently than in other files running on the backend, maybe it has something to do with the Webpack lifecycle and when exactly those markdoc files are included,

I'll try to debug this further.

thanks for replying!
